src/daft-core/src/datatypes/logical.rs

```diff
@@ -1,121 +1,153 @@
+        //     physical.data_type()
+        // );
+        LogicalArrayImpl {
             physical,
             field,
             marker_: PhantomData,
         }
     }
 
-    pub fn empty(name: &str, dtype: &DataType) -> Self {
-        let field = Field::new(name, dtype.clone());
-        Self::new(field, DataArray::empty(name, &dtype.to_physical()))
-    }
-
     pub fn name(&self) -> &str {
         self.field.name.as_ref()
     }
 
-    pub fn rename(&self, name: &str) -> Self {
-        let new_field = self.field.rename(name);
-        let new_array = self.physical.rename(name);
-        Self::new(new_field, new_array)
-    }
-
     pub fn field(&self) -> &Field {
         &self.field
     }
 
-    pub fn logical_type(&self) -> &DataType {
+    pub fn data_type(&self) -> &DataType {
         &self.field.dtype
     }
+}
 
-    pub fn physical_type(&self) -> &DataType {
-        self.physical.data_type()
-    }
-
-    pub fn len(&self) -> usize {
-        self.physical.len()
-    }
-
-    pub fn is_empty(&self) -> bool {
-        self.len() == 0
-    }
+macro_rules! impl_logical_type {
+    ($physical_array_type:ident) => {
+        pub fn len(&self) -> usize {
+            self.physical.len()
+        }
 
-    pub fn size_bytes(&self) -> DaftResult<usize> {
-        self.physical.size_bytes()
-    }
+        pub fn is_empty(&self) -> bool {
+            self.len() == 0
+        }
 
-    pub fn slice(&self, start: usize, end: usize) -> DaftResult<Self> {
-        let new_array = self.physical.slice(start, end)?;
-        Ok(Self::new(self.field.clone(), new_array))
-    }
+        pub fn empty(field_name: &str, dtype: &DataType) -> Self {
+            let physical = $physical_array_type::empty(field_name, &dtype.to_physical());
+            let field = Field::new(field_name, dtype.clone());
+            Self::new(field, physical)
+        }
 
-    pub fn head(&self, num: usize) -> DaftResult<Self> {
-        self.slice(0, num)
-    }
+        pub fn concat(arrays: &[&Self]) -> DaftResult<Self> {
+            if arrays.is_empty() {
+                return Err(common_error::DaftError::ValueError(
+                    "Need at least 1 logical array to concat".to_string(),
+                ));
+            }
+            let physicals: Vec<_> = arrays.iter().map(|a| &a.physical).collect();
+            let concatd = $physical_array_type::concat(physicals.as_slice())?;
+            Ok(Self::new(arrays.first().unwrap().field.clone(), concatd))
+        }
+    };
+}
 
-    pub fn concat(arrays: &[&Self]) -> DaftResult<Self> {
-        if arrays.is_empty() {
-            return Err(common_error::DaftError::ValueError(
-                "Need at least 1 logical array to concat".to_string(),
-            ));
+/// Implementation for a LogicalArray that wraps a DataArray
+impl<L: DaftLogicalType> LogicalArrayImpl<L, DataArray<L::PhysicalType>> {
+    impl_logical_type!(DataArray);
+
+    pub fn to_arrow(&self) -> Box<dyn arrow2::array::Array> {
```

Comment:
This code was ported directly from our old `impl SeriesLike for LogicalArray`, but I think it can be greatly simplified. Probably just call `to_arrow` on self.physical and then change the arrow type to the logical arrow type.